### PR TITLE
Redesign Wallet UI

### LINKED ISSUE #3408: [UI/UX Proposal] Wallet View V2
Continuing to flatten the UI navigation pattern, the latest proposal to improve the Wallet view, as discussed in the design meeting:

I've got a few things I wanted to polish up, as there are parts of this that are still quite rough, but posting for transparency, feedback, and iterative progress as we start to build.

TODO:
- The only navigation issue at the time of posting is how the user navigates back to the "transaction" view, but this is an easy issue and will be resolved.
- Mobile View mocks
- Fixed header section (balance, send/receive, settings) on scroll .. to match other patterns

GIF Clickthrough (had to compress the hell out of it to post, might record a smoother one on my other machine):
TW_LOOK_02

Static assets, not to be heavily scrutinized yet, but posting for details and progress:
WalletV2Transactions
WalletV2Settings
WalletV2Receive
WalletV2Send
WalletV2Rescan

### PR DESCRIPTION
**Pull request by dennisreimann**
Closes #3408. Mostly finetuning is left. Opening as draft so that we can discuss it already.

image

### CODE REVIEW

**Comment by dennisreimann** [comment]
Integrated CPFP. One remark, as this is now part of the actions menu, which looks like this currently:

image

Should we move the wallet related actions to the wallet settings and make them available there instead here in the actions menu? I think it would make the distinction between transaction and wallet related actions clearer and the transaction actions menu would work just as the invoice actions menu - tying the actions to the whole list/selected items.

I also think the wallet related actions aren't used frequently, hence they'd better fit into the settings. As we have the rescan pointer at the bottom of the list, users that would miss their transactions still would see a prominent link to that action.

/cc @pavlenex @NicolasDorier @dstrukt

**Comment by dstrukt** [comment]
I agree with that assessment.

File: BTCPayServer/wwwroot/main/layout.css [hunk 1]
```diff
@@ -19,7 +19,7 @@
     --icon-size: 1.5rem;
     
     height: var(--header-height);
-    z-index: 9999999;
```
Comment (dennisreimann) [inline BTCPayServer/wwwroot/main/layout.css:22]: That value caused the main menu to be above the modal overlay.

File: BTCPayServer/Components/MainNav/MainNav.cs [hunk 2]
```diff
@@ -19,7 +19,6 @@ namespace BTCPayServer.Components.MainNav
 {
     public class MainNav : ViewComponent
     {
-        private const string RootName = "Global";
```
Comment (dennisreimann) [inline BTCPayServer/Components/MainNav/MainNav.cs:22]: Cleaning up old leftovers.

File: BTCPayServer/Components/WalletNav/Default.cshtml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+@using BTCPayServer.Client
+@using BTCPayServer.Views.Wallets
+@using BTCPayServer.Abstractions.Extensions
+@inject BTCPayNetworkProvider _btcPayNetworkProvider
+
+@model BTCPayServer.Components.WalletNav.WalletNavViewModel
+@addTagHelper *, BundlerMinifier.TagHelpers
+
+<div class="d-sm-flex align-items-center justify-content-between">
+    <a asp-controller="UIWallets" asp-action="WalletTransactions" asp-route-walletId="@Model.WalletId" class="unobtrusive-link">
```
Comment (dennisreimann) [inline BTCPayServer/Components/WalletNav/Default.cshtml:11]: @dstrukt that link might not be the most obvious, but I found it to be intuitive and useful: We're linking the wallet name and balance back to the transaction list here.

File: BTCPayServer/Services/Labels/LabelFactory.cs [hunk 4]
```diff
@@ -90,5 +92,14 @@ private ColoredLabel CreateLabel(LabelData uncoloredLabel, string color, HttpReq
             }
             return coloredLabel;
         }
+        
+        private string TextColor(string bgColor)
```
Comment (dennisreimann) [inline BTCPayServer/Services/Labels/LabelFactory.cs:96]: Calculating the text color based on the label background color. The text will be either black or white, depending on the lightness of the background color.

File: BTCPayServer/wwwroot/js/copy-to-clipboard.js [hunk 5]
```diff
@@ -10,8 +10,8 @@ window.copyToClipboard = function (e, data) {
     const item = e.target.closest('[data-clipboard]');
     const confirm = item.querySelector('[data-clipboard-confirm]') || item;
     const message = confirm.getAttribute('data-clipboard-confirm') || 'Copied ✔';
-    if (!confirm.dataset.clipboardInitialText) {
-        confirm.dataset.clipboardInitialText = confirm.innerText;
+    if (!confirm.dataset.clipboardInitial) {
+        confirm.dataset.clipboardInitial = confirm.innerHTML;
```
Comment (dennisreimann) [inline BTCPayServer/wwwroot/js/copy-to-clipboard.js:14]: Has to be `innerHTML` now, as we are using an SVG as the button content.

**Review by bolatovumar — APPROVED** [review]
Looks good. One minor issue that the transactions list is wider than the fixed header so when you scroll you can see the transactions list behind the header:

wallet

**Comment by dennisreimann** [comment]
@bolatovumar good point, I'd like to make it span the full content width and potentially enhance it with a shadow indicating the scroll status. Will do that in a separate PR.

**Comment by dennisreimann** [comment]
@bolatovumar Adressed this in #347, see [6398cac](https://github.com/btcpayserver/btcpayserver/pull/3471/commits/6398cac6f69c5246d4ba6f41a28739fec7af8967).

**Pull request merged**